src/daft-local-execution/src/dispatcher.rs

```diff
@@ -1,74 +1,175 @@
 use std::sync::Arc;
 
-use async_trait::async_trait;
 use common_error::DaftResult;
 use daft_dsl::ExprRef;
+use daft_micropartition::MicroPartition;
 
 use crate::{
-    buffer::RowBasedBuffer, channel::Sender, pipeline::PipelineResultType,
+    buffer::RowBasedBuffer,
+    channel::{create_channel, Receiver, Sender},
     runtime_stats::CountingReceiver,
+    ExecutionRuntimeHandle,
 };
 
-#[async_trait]
+/// The `Dispatcher` trait defines the interface for dispatching morsels to workers.
+///
+/// The `dispatch` method takes:
+/// - A vector of input receivers
+/// - The number of workers
+/// - A runtime handle for the current execution context
+/// - The name of the operator
+///
+/// It returns a vector of receivers (one per worker) that will receive the morsels.
+///
+/// Implementations must spawn a task on the runtime handle that reads from the
```

Comment:
I think it would make more sense to name this:
```
trait DispatcherBuilder {
  fn spawn_dispatcher(...) -> SpawnedDispatcherResult;
}

struct SpawnedDispatcherResult {
  receivers:  Vec<Receiver<Arc<MicroPartition>>>,
  dispatch_handle: SpawnedTask
}
```

I think my key confusion was that `dispatch` sounds like an action that would happen on every tick rather than something that happens once.

Reply:
Gotcha, made it more explicit that it should _spawn_ a dispatching task. Also implemented the `SpawnedDispatcherResult`, then the caller can await on it themselves.